File: app/src/test/java/com/example/pam_project/landing/WelcomeActivityTest.java
package com.example.pam_project.landing;

import com.example.pam_project.R;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.android.controller.ActivityController;

import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class WelcomeActivityTest {

    private ActivityController<WelcomeActivity> activity;

    @Before
    public void setup() {
        activity = Robolectric.buildActivity(WelcomeActivity.class);
    }

    @Test
    public void givenWelcomeWasConfirmedThenVerifyTheActivityIsFinished() {
        activity.create().resume();

        activity.get().findViewById(R.id.welcome_button).performClick();

        assertTrue(activity.get().isFinishing());
    }
}

File: app/src/test/java/com/example/pam_project/landing/WelcomeActivityTest.kt
package com.example.pam_project.landing

import android.view.View
import com.example.pam_project.R
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner
import org.robolectric.android.controller.ActivityController

@RunWith(RobolectricTestRunner::class)
class WelcomeActivityTest {
    private lateinit var activity: ActivityController<WelcomeActivity>
    @Before
    fun setup() {
        activity = Robolectric.buildActivity(WelcomeActivity::class.java)
    }

    @Test
    fun givenWelcomeWasConfirmedThenVerifyTheActivityIsFinished() {
        activity.create().resume()
        activity.get().findViewById<View>(R.id.welcome_button).performClick()
        Assert.assertTrue(activity.get().isFinishing)
    }
}